### PR TITLE
#1994 - [Foundation Elimination] Modals [Phase III] 

### PR DESCRIPTION
**Pull request by gibbsfromncis**
All screenshots can be find there - https://github.com/bitshares/bitshares-ui/issues/1994

@sschiessl-bcp I did a lot of tests but I suppose the thing you should test it is that Modals open/close nice because in most cases I didn't touch functionality inside Modals.

### CODE REVIEW

**Comment by startailcoon** [comment]
Does this fix https://github.com/bitshares/bitshares-ui/issues/2039?

File: app/components/PrivateKeyView.jsx [hunk 1]
```diff
@@ -16,21 +16,49 @@ export default class PrivateKeyView extends Component {
...
+        this.hideQrModal = this.hideQrModal.bind(this);
+
+        this.onClose = this.onClose.bind(this);
     }
 
     _getInitialState() {
-        return {wif: null};
+        return {
+            isModalVisible: false,
+            isQrModalVisible: false,
```
Comment (sschiessl-bcp) [inline app/components/PrivateKeyView.jsx]: The more I read this PR the more I am wondering if the Modal can't handle the visibility by itself without parent state, and only do on button click callbacks where necessary.

Was this evaluated? If so, was there anything against such an approach?

Comment (sschiessl-bcp) [inline app/components/PrivateKeyView.jsx]: The more I read this PR the more I am wondering if the Modal can't handle the visibility by itself without parent state, and only do on button click callbacks where necessary.

Was this evaluated? If so, was there anything against such an approach?

Comment (sschiessl-bcp) [inline app/components/PrivateKeyView.jsx]: The more I read this PR the more I am wondering if the Modal can't handle the visibility by itself without parent state, and only do on button click callbacks where necessary.

Was this evaluated? If so, was there anything against such an approach?

File: app/components/Account/AccountPortfolioList.jsx [hunk 2]
```diff
@@ -37,8 +37,12 @@ class AccountPortfolioList extends React.Component {
         super();
 
         this.state = {
+            isBridgeModalVisible: false,
```
Comment (sschiessl-bcp) [inline app/components/Account/AccountPortfolioList.jsx]: The more I read this PR the more I am wondering if the Modal can't handle the visibility by itself without parent state, and only do on button click callbacks where necessary.

Was this evaluated? If so, was there anything against such an approach?

**Comment by sschiessl-bcp** [comment]
Sorry if anyone got spammed here .. github had some issues ... will try to review now

**Pull request merged**